Lib/inspect.py

```diff
@@ -1821,8 +1814,10 @@ def getattr_static(obj, attr, default=_sentinel):
        documentation for details.
     """
     instance_result = _sentinel
-    if not _is_type(obj):
-        klass = type(obj)
+
+    objtype = type(obj)
+    if type not in _static_getmro(objtype):
+        klass = objtype
```

Comment:
This has equivalent semantics with less code and fewer assignments, but perhaps it muddies the semantics of the `klass` var a bit (since it will temporarily be "wrong" in the case that `obj` is already a type.) No strong feelings, whatever you prefer.

```suggestion
    klass = type(obj)
    if type not in _static_getmro(klass):
```

Reply:
Hmm. Yeah, I see what you mean, but I think I'd prefer to keep the semantics of `klass` clear here :)